samples/boards/mimxrt1170_evk_cm7/magic_addr/src/main.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/linker/section_tags.h>
+#include "memc_nxp_flexram.h"
+#include <zephyr/kernel.h>
+#include <zephyr/console/console.h>
+#include <zephyr/cache.h>
+
+#define DTCM_NODE DT_NODELABEL(dtcm)
+#define DTCM_BASE DT_REG_ADDR(DTCM_NODE)
+
+K_SEM_DEFINE(dtcm_magic, 0, 1);
+
+__dtcm_bss_section uint8_t var;
+int cnt;
+
+void flexram_magic_addr_isr_cb(enum memc_flexram_interrupt_cause cause,
+						void *data)
+{
+	ARG_UNUSED(data);
+
+	if (cause == flexram_dtcm_magic_addr) {
+		printf("Magic DTCM address accessed %d times\n", ++cnt);
+		k_sem_give(&dtcm_magic);
+	}
+}
+
+
+int main(void)
+{
+	sys_cache_data_disable();
+
+	memc_flexram_register_callback(flexram_magic_addr_isr_cb, NULL);
+
+	console_init();
+
+	printf("%s is opening spellbook...\n", CONFIG_BOARD);
```

Comment:
While I appreciate the humor, I don't think this sample really helps the user understand what is going on here- based on the RM, it seems what should occur here is that when the user presses a key, the `var` value in DTCM will be written to, and the flexRAM controller will issue an interrupt (as `var` is set to the magic address)

If this is all correct, I'd like to see documentation in the sample making it clear that this is what is occurring, since the currently readme/print statements don't make this obvious

Reply:
I added more print statements and README info